### PR TITLE
Node/EVM: More nodes support finalized and safe

### PR DESCRIPTION
**Pull request by bruce-riley**
Due to recent chain updates, the following chains now support querying for `finalized` and `safe` blocks:

- Acala
- Arbitrum
- BSC
- Karura
- Moonbeam

This is on top of the chains that already did:

- Ethereum / Goerli / Sepolia
- Base
- Optimism

This PR also streamlines the watcher setup code to make it easier to read and maintain.

### CODE REVIEW

File: node/pkg/watchers/evm/watcher.go [hunk 1]
```diff
@@ -661,7 +592,7 @@ func (w *Watcher) Run(parentCtx context.Context) error {
 
 				start := time.Now()
 				currentHash := ev.Hash
-				logger.Debug("processing new header",
+				logger.Info("processing new header",
```
Comment (evan-gray) [inline node/pkg/watchers/evm/watcher.go:595]: Do we want to keep this as Info? feel like that will cause some log spam

File: node/pkg/watchers/evm/watcher.go [hunk 2]
```diff
@@ -197,16 +198,10 @@ func (w *Watcher) Run(parentCtx context.Context) error {
...
-		ufb, err := w.getAcalaMode(ctx)
-		if err != nil {
-			return err
-		}
-
-		if ufb {
-			useFinalizedBlocks = true
-		}
+	var useFinalizedBlocks, safeBlocksSupported bool
+	useFinalizedBlocks, safeBlocksSupported, err = w.getFinality(ctx)
```
Comment (evan-gray) [inline node/pkg/watchers/evm/watcher.go:202]: nice.

**Review by evan-gray — APPROVED** [review]
Want to remove the custom finalizers which are no longer used as well?

File: node/pkg/watchers/evm/watcher.go (hunk 1 above)
Comment (bruce-riley) [reply node/pkg/watchers/evm/watcher.go]: Doh! Didn't mean to commit that. Changed it back.

File: node/pkg/watchers/evm/watcher.go [hunk 3]
```diff
@@ -889,41 +820,56 @@ func fetchCurrentGuardianSet(ctx context.Context, ethConn connectors.Connector)
...
-func (w *Watcher) getAcalaMode(ctx context.Context) (useFinalizedBlocks bool, errRet error) {
-	timeout, cancel := context.WithTimeout(ctx, 15*time.Second)
-	defer cancel()
-
-	c, err := rpc.DialContext(timeout, w.url)
-	if err != nil {
-		errRet = fmt.Errorf("failed to connect to url %s to check acala mode: %w", w.url, err)
-		return
+// getFinality determines if the chain supports "finalized" and "safe". This is hard coded so it requires thought to change something. However, it also reads the RPC
+// to make sure the node actually supports the expected values, and returns an error if it doesn't. Note that we do not support using safe mode but not finalized mode.
```
Comment (panoel) [inline node/pkg/watchers/evm/watcher.go:824]: The last sentence "Note....." is confusing.

**Review by panoel — APPROVED** [review]

**Review by evan-gray — APPROVED** [review]

**Pull request merged**